test/serializers/attributes_test.rb

```diff
@@ -18,11 +18,31 @@ def test_attributes_definition
           @profile_serializer.class._attributes)
       end
 
+      def test_attributes
+        assert_equal({ name: 'Name 1', description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
       def test_attributes_with_fields_option
         assert_equal({ name: 'Name 1' },
           @profile_serializer.attributes(fields: [:name]))
       end
 
+      def test_attributes_with_fields_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
+          keys - [:name]
+        end
+        assert_equal({ description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
+      def test_attributes_with_fields_option_and_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
```

Comment:
I don't think I'm quite following you. Granted, I'm not very familiar with the code base so I could be missing something. 

There are tests verify the behavior of the introduced method method `fields`. This is pretty much covered by existing tests. Also the output of `attributes` without overwritten method and options is tested. I'm quite sure that mutations would be catched by the tests. 

Is there serializer classes that would overwrite the default behaviour? Or some cases where you'd think an integration test would be needed?